dev/com.ibm.ws.security.acme/src/com/ibm/ws/security/acme/internal/AcmeProviderImpl.java

```diff
@@ -873,5 +911,98 @@ protected void updateAcmeConfigService(AcmeConfigService acmeConfigService, Map<
 	public void setAcmeApplicationStateListener(AcmeApplicationStateListener acmeApplicationStateListener) {
 		applicationStateListenerRef.set(acmeApplicationStateListener);
 	}
+	
+	/**
+	 * This method determines if the directoryURI has changed by checking
+	 * the acme file.
+
+	 * @return True if the directoryURI has changed
+	 * @throws AcmeCaException
+	 */
+	private boolean directoryURIChanged() throws AcmeCaException {
```

Comment:
I think I would put all the logic related to historical records into its own class and call it from AcmeProviderImpl.